lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,309 @@
+                     if (newSessionId) {
+                         enrollText = currentSessionId ? gettext('Change Session') : gettext('Select Session');
+                     } else {
+                         enrollText = gettext('Leave Current Session');
+                     }
+                     this.$('.enroll-btn-initial').text(enrollText);
+
+                     // Update the button popover text to enable two step authentication.
+                     if (newSessionId) {
+                         confirmationHTML = currentSessionId ?
+                             gettext('Are you sure that you would like to switch session?') + '<br><br>' +
+                             gettext(' Please know that by choosing to switch session you will lose any progress you' +
+                             'have made in this session.') :
+                             gettext('Are you sure that you would like to select this session?');
+                     } else {
+                         confirmationHTML = gettext('Are you sure that you would like to leave this session?') +
+                             '<br><br>' + gettext('Please know that by leaving you will lose any progress you had' +
+                             ' made in this session.');
+                     }
+
+                     // Remove existing popover and re-initialize
+                     enrollBtnInitial.popover('dispose');
+                     enrollBtnInitial.popover({
+                         placement: 'bottom',
+                         container: this.$el,
+                         html: true,
+                         trigger: 'click',
+                         content: '<div class="verification-modal" role="dialog"' +
+                             'aria-labelledby="enrollment-verification-title">' +
+                             '<p id="enrollment-verification-title">' + confirmationHTML + '</p>' +
+                             '<div class="action-items">' +
+                             '<button type="button" class="popover-dismiss final-confirmation-btn">' +
+                             gettext('No') + '</button>' +
+                             '<button type="button" class="enroll-btn final-confirmation-btn">' +
+                             gettext('Yes') + '</button></div></div>'
+                     });
+
+                     // Close popover on click-away
+                     $(document).on('click', function(e) {
+                         if (!($(e.target).closest('.enroll-btn-initial, .popover').length)) {
+                             this.$('.enroll-btn-initial').popover('hide');
+                         }
+                     }.bind(this));
+
+                     // Ensure that focus moves to the popover on click of the initial change enrollment button.
+                     this.$('.enroll-btn-initial').on('click', function() {
+                         this.$('.final-confirmation-btn:first').focus();
+                     }.bind(this));
```

Comment:
I'm slightly nervous about these event handlers being instantiated multiple times if ```updateEnrollBtn``` is called more than once.

Reply:
Moved it to a postRender function so it gets initialized only once